runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java

```diff
@@ -118,16 +123,18 @@ public KafkaClientConnectionPool(
     {
         this.proxyTypeId = context.supplyTypeId("proxy");
         this.writeBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
-        this.extBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
+        this.encodeBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
         this.bufferPool = context.bufferPool();
         this.signaler = context.signaler();
+        this.connectionSignaler = new KafkaClientSignaler();
```

Comment:
Recommend passing regular signaler to `KafkaClientSignaler` constructor, then call this local field `signaler` instead of `connectionSignaler`.